pkg/middleware/auth.go

```
@@ -99,6 +102,99 @@ func (c JWTVerifier) CheckToken(token string) (*jwt.Token, error) {
 	return nil, lastSeenErr
 }
 
+// JWTVerifierURL checks token with JWT against JWKS, implements Verifier
+type JWTVerifierURL struct {
+	jwksURL string
+
+	parser   *jwt.Parser
+	jwksKeys *jwk.Set
+	jwksLock sync.RWMutex
+}
+
+func (c *JWTVerifierURL) startTicker(ticker time.Duration) {
+
+	tick := time.NewTicker(ticker)
+	defer tick.Stop()
+
+	for range tick.C {
+		err := c.updateKeySet()
+		if err != nil {
+			log.Warn().Err(err).Msg("unable to update JWKS key set")
+		}
+	}
+}
+
+func (c *JWTVerifierURL) updateKeySet() error {
+
+	c.jwksLock.Lock()
+	defer c.jwksLock.Unlock()
+
+	set, err := jwk.Fetch(c.jwksURL)
+	if err != nil {
+		return err
+	}
+	c.jwksKeys = set
+	return nil
+}
+
+func (c *JWTVerifierURL) getKeySet() *jwk.Set {
+	c.jwksLock.RLock()
+	defer c.jwksLock.RUnlock()
+	return c.jwksKeys
+
+}
+
+// NewJWTVerifierURL creates JWTVerifierURL with JWKS URL
+func NewJWTVerifierURL(jwksURL string, updateInterval time.Duration) *JWTVerifierURL {
+
+	http.DefaultClient = &http.Client{Timeout: 10 * time.Second}
+	jwtVerifierURL := JWTVerifierURL{jwksURL: jwksURL, parser: new(jwt.Parser)}
+	err := jwtVerifierURL.updateKeySet()
+
+	if err != nil {
+		log.Error().Msg("unable to construct NewJWTVerifierURL")
```

We're swallowing the error here. Should be:
```go
log.Error().Err(err).Msg("unable to fetch key set")
```